### ISSUE TITLE
New Rule: focus-and-hover

### ISSUE THREAD

**Issue by Andersos** [sasstools/sass-lint#349]
Because of usability it is usually [good](http://thatjdanisso.cool/a11y/focus-vs-hover/) to apply the same rules to both `:hover` and `:focus`. Maybe we could add a rule to detect if we forget one of the pseudo-classes (usually it would be `:focus`).

When enabled, the following are allowed:

``` css
a {
  text-decoration: none;
  &:hover,
  &:focus {
    text-decoration: underline;
}
```

When enabled, the following are disallowed:

``` css
a {
  text-decoration: none;
  &:hover {
    text-decoration: underline;
}
```

**Comment by DanPurdy** [comment]
I think this makes sense personally, it's a nice usability win and should probably make its way in. Only area to think about is when you actually want to specify separate focus and hover styles which is rare I know but could be possible.

**Comment by bmds** [comment]
An alternative approach that should be more robust is to require both defined within the parent block if one is used in a declaration.

For example:
```sass
.first {
  text-decoration: none;
 
  &:hover {
    text-decoration: underline;
  }
}
.second {
  text-decoration: none;
 
  &:focus {
    text-decoration: underline;
  }
}
```
Would both fail as they have only got one of the pseudo-classes. The following would both pass:

```sass
.first {
  text-decoration: none;
 
  &:hover,
  &:focus {
    text-decoration: underline;
  }
}
.second {
  text-decoration: none;
 
  &:hover {
    color: red;
  }
 
  &:focus {
    text-decoration: underline;
  }
}
```

This enforces accessibility concerns but allows different styling. Another consideration would be :active which is also frequently missed.

**Comment by DanPurdy** [comment]
Yeah this is how I envisaged having it work after thinking about it a bit more the other day. I think it would be a nice addition to the rulesets. Enforcing hover/focus and also active should probably be 2 separate rules though with our aim of small single purpose rulesets. 

Thanks guys will look into it.

**Comment by bgriffith** [comment]
Had a look at this today and have got a working version. One thing that I think we need to decide is what to do in the following situation where we have the same selector;

```scss
.baz {
    &:focus {
      color: #333;
      text-decoration: underline;
    }
  }
  .baz {
    &:hover {
      color: #333;
    }
  }
```
Should that be flagged as an issue or not? I'm torn.

**Comment by DanPurdy** [comment]
IMO no it shouldn't be flagged it's valid the fact that the selectors aren't merged should be left up to another rule. Unless it's something else you aren't sure about.

**Comment by Andersos** [comment]
According to this rule they have both focus and hover on the baz class.

**Comment by jameelmoses** [comment]
What's the status of this? Anything I can do to help?

**Comment by Andersos** [comment]
Looks like @bgriffith has made an attempt to solve this. Don't know the status of this other than that.

**Comment by DanPurdy** [comment]
I've pinged @bgriffith and will see what the status is, I'm guessing it's long since disappeared the version of this though. Anyone is obviously welcome to drop a PR in in the meantime and beat him to the punch! 😄
